design-documents/provisioning/configuration.md

```diff
@@ -64,26 +81,174 @@ The usual behavior for installing from scratch or upgrading is the following:
+  * new entity is installed
+  * checkbox does not exist
+  * admin can't change the entity
+  * will overwrite admin side manual created entities
+
+#### Example: Attribute behavior
+ - "**is_mutable**" flag is checkbox at the attribute grid and at attribute edit page
+ - Attributes which created from Admin panel is not maintainable by xml (no dumping, "**is_mutable**" flag is not shown but has true value in database) until xml with the same attribute code will be created and "**is_mutable**" will be unchecked
+ - "**is_mutable**" flag is responsible for defining maintenance responsibilities:
+   - **true** responsibility on Admin panel side switchable from admin
+   - **false** responsibility on code side switchable from admin
+   - **locked** - responsibility on code side and not switchable from admin
+```
+<?xml version="1.0"?>
+<!--
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+-->
+<attributes xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="urn:magento:framework:Attributes/Declaration/Data/etc/attributes.xsd">
+    <customer>
+        <attribute xsi:type="date"
+           code="custom_design_from"
+           frontend_label="Active From"
+           backend_type="datetime"
+           is_required="false"
+           is_user_defined="true"
+           is_unique="false"
+           backend_model="Magento\Catalog\Model\Attribute\Backend\Startdate"
+           attribute_model="Magento\Catalog\Model\ResourceModel\Eav\Attribute"
+           is_mutable="false" <!-- is optional and has default value false-->/>
+    </customer>
+</attributes>
+```
+## Created by XML
+- Default `"is_mutable"=false` and this property is optional for attribute xml node
+- Fully managed by admin panel
+- If xml with the same attribute code will be created - it will take over all responsibilities to a code side and will be then managed like attribute created by XML declaration
+
+| is_mutable in XML->       | `is_mutable` not defined          | `is_mutable="false"` |
+| ------------------------- | --------------------------------- | -------------------- |
+| DB value for `is_mutable` | true/false (default false)        | locked               |
+| checkbox                  | unchecked                         | not existing         |
+| checkbox changeable       | true                              | not existing         |
+| save btn                  | disabled (until checkbox checked) | disabled             |
+| delete btn                | disabled (until checkbox checked) | disabled             |
+| data patch changes        | can apply                         | will be overwritten  |
+
+
+## Created by admin (xml declaration doesn't exists):
+- Fully managed by admin panel
+- If xml with the same attribute code will be created - it will take over all responsibilities to a code side and will be then managed like attribute created by XML declaration
```

Comment:
Maybe this should be tru only for non-mutable entities? Though you may provide more real-world scenarios to understand what would be more convenient.
For me described behavior makes things more complex. Like, if I'm looking at an instance (installed by somebody some time ago), how do I know if it's a feature or a bug that some values are taken from files, and others are not (probably Admin entered them). I would prefer to not rely on time for the application logic as it's difficult to reason about.